test/Microsoft.VisualStudio.Threading.Tests/JoinableTaskContextTests.cs

```diff
@@ -45,6 +45,55 @@ public void IsWithinJoinableTask()
         });
     }
 
+    [Fact]
+    public void IsMainThreadBlockedByAnyone()
+    {
+        Assert.False(this.Context.IsMainThreadBlockedByAnyone);
+        AsyncManualResetEvent mainThreadBlockerEvent = new AsyncManualResetEvent(false);
+        AsyncManualResetEvent backgroundThreadMonitorEvent = new AsyncManualResetEvent(false);
+
+        // Start task to monitor IsMainThreadBlockedByAnyone
+        Task.Run(async () =>
+        {
+            await TaskScheduler.Default.SwitchTo(alwaysYield: true);
+            await mainThreadBlockerEvent.WaitAsync();
+
+            bool shouldWaitForMainThreadBlock = true;
+            while (shouldWaitForMainThreadBlock)
+            {
+                try
+                {
+                    Assert.True(this.Context.IsMainThreadBlockedByAnyone);
+                    shouldWaitForMainThreadBlock = false;
+                }
+                catch
+                {
+                    // Give the main thread time to enter a blocking state
+                    Thread.Sleep(50);
+                }
+            }
+
+            backgroundThreadMonitorEvent.Set();
+        });
+
+        JoinableTask? joinable = this.Factory.RunAsync(async delegate
+        {
+            Assert.False(this.Context.IsMainThreadBlockedByAnyone);
+            await Task.Yield();
+
+            this.Factory.Run(async () =>
+            {
+                await TaskScheduler.Default.SwitchTo(alwaysYield: true);
+                mainThreadBlockerEvent.Set();
+                await backgroundThreadMonitorEvent.WaitAsync();
```

Comment:
This could hang the test if the event is never raised. Use a cancellation token.
```suggestion
                await backgroundThreadMonitorEvent.WaitAsync(this.TimeoutToken);
```